### PR TITLE
Study hub/onboarding overlay

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

Implemented a semi generic onboarding overlay component used for the study hub interest onboarding.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

## Screenshots

Simulator Screen Shot - iPhone 13 - 2021-11-23 at 11 22 24

https://user-images.githubusercontent.com/17004429/143598644-a2912fcc-d065-4b61-9572-0e107090fefa.mp4

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Checklist (for merger)

- [ ] Make sure the right merge strategy is chosen, commit merge into master and squash merge into develop.

### CODE REVIEW

File: src/core/state/store.ts [hunk 1]
```diff
@@ -2,10 +2,10 @@ import rootReducer from '@covid/core/state/root';
 import AsyncStorage from '@react-native-async-storage/async-storage';
 import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
 import { LogBox } from 'react-native';
-import { FLUSH, PAUSE, PERSIST, persistReducer, persistStore, PURGE, REGISTER, REHYDRATE } from 'redux-persist';
+import { persistReducer, persistStore } from 'redux-persist';
 
 const persistConfig = {
-  blacklist: ['app'],
+  blacklist: ['app', 'onboarding'],
```
Comment (juliatan) [inline src/core/state/store.ts:8]: does this mean you don't want to persist the onboarding state? if so, why not?

File: src/features/study-hub/components/RowInterested.tsx [hunk 2]
```diff
@@ -78,10 +120,46 @@ export function RowInterested(props: TProps) {
     } catch (_) {}
 
     setLoading(false);
-  }, [props.study.patient_is_interested, startupInfo?.show_study_hub_interest_onboarding]);
+  }, [
+    dispatch,
```
Comment (juliatan) [inline src/features/study-hub/components/RowInterested.tsx:124]: does dispatch really need to be a dependency?

File: src/core/state/store.ts (hunk 1 above)
Comment (aikewoody) [reply src/core/state/store.ts:8]: The values we store in this redux slice are only needed during runtime. Another reason for blacklisting is that the `renderIcon` value can't be serialised and the `iconXValue` and `iconYValue` could potentially be different between app runs so better to start with a clean state

File: src/features/study-hub/components/RowInterested.tsx (hunk 2 above)
Comment (aikewoody) [reply src/features/study-hub/components/RowInterested.tsx:124]: Yes, the ESLint rule `react-hooks/exhaustive-deps`enforces this (we have it currently disabled but it should be enabled again)

**Pull request merged**